modules/ppcp-api-client/src/Entity/FraudProcessorResponse.php

```diff
@@ -64,11 +64,99 @@ public function cvv_code(): ?string {
 	 */
 	public function to_array(): array {
 		return array(
-			'avs_code'      => $this->avs_code() ?: '',
+			'avs_code'      => $this->avs_code(),
+			'cvv2_code'     => $this->cvv_code(),
+			// For backwards compatibility.
 			'address_match' => $this->avs_code() === 'M' ? 'Y' : 'N',
 			'postal_match'  => $this->avs_code() === 'M' ? 'Y' : 'N',
 			'cvv_match'     => $this->cvv_code() === 'M' ? 'Y' : 'N',
 		);
 	}
 
+	/**
+	 * Retrieves the AVS (Address Verification System) code messages based on the AVS response code.
+	 *
+	 * Provides human-readable descriptions for various AVS response codes
+	 * and returns the corresponding message for the given code.
+	 *
+	 * @return string The AVS response code message. If the code is not found, an error message is returned.
+	 */
+	public function get_avs_code_message(): string {
+		if ( ! $this->avs_code() ) {
+			return '';
+		}
+		$messages = array(
+			/* Visa, Mastercard, Discover, American Express */
+			'A' => 'A: Address - Address only (no ZIP code)',
+			'B' => 'B: International "A" - Address only (no ZIP code)',
+			'C' => 'C: International "N" - None. The transaction is declined.',
+			'D' => 'D: International "X" - Address and Postal Code',
+			'E' => 'E: Not allowed for MOTO (Internet/Phone) transactions - Not applicable. The transaction is declined.',
+			'F' => 'F: UK-specific "X" - Address and Postal Code',
+			'G' => 'G: Global Unavailable - Not applicable',
+			'I' => 'I: International Unavailable - Not applicable',
+			'M' => 'M: Address - Address and Postal Code',
+			'N' => 'N: No - None. The transaction is declined.',
+			'P' => 'P: Postal (International "Z") - Postal Code only (no Address)',
+			'R' => 'R: Retry - Not applicable',
+			'S' => 'S: Service not Supported - Not applicable',
+			'U' => 'U: Unavailable / Address not checked, or acquirer had no response. Service not available.',
+			'W' => 'W: Whole ZIP - Nine-digit ZIP code (no Address)',
+			'X' => 'X: Exact match - Address and nine-digit ZIP code)',
+			'Y' => 'Y: Yes - Address and five-digit ZIP',
+			'Z' => 'Z: ZIP - Five-digit ZIP code (no Address)',
+			/* Maestro */
+			'0' => '0: All the address information matched.',
+			'1' => '1: None of the address information matched. The transaction is declined.',
+			'2' => '2: Part of the address information matched.',
+			'3' => '3: The merchant did not provide AVS information. Not processed.',
+			'4' => '4: Address not checked, or acquirer had no response. Service not available.',
+		);
+
+		/**
+		 * Psalm suppress
+		 *
+		 * @psalm-suppress PossiblyNullArrayOffset
+		 * @psalm-suppress PossiblyNullArgument
+		 */
+		return $messages[ $this->avs_code() ] ?? sprintf( '%s: Error', $this->avs_code() );
```

Comment:
if the constructor param `?string $avs_code` is `'null'` or `''` then `$this->avs_code()` here will return `''` and the error message will be like this: `string(7) ": Error"`

Reply:
That will be checked in Line 85.